src/main/kotlin/no/nav/familie/ef/søknad/service/OppslagServiceServiceImpl.kt

```diff
@@ -1,83 +1,117 @@
+        throw ApiFeil("Personer med tilknytning til søker har adressesperre", HttpStatus.FORBIDDEN)
     }
 
-    private fun hentSøkerinfoFraTPSDto(): Søkerinfo {
-        val personinfoDto = client.hentPersoninfo()
-        val barn = client.hentBarn()
-        val aktuelleBarn = barn.filter { erIAktuellAlder(it.fødselsdato) }
-        return settNavnFraPdlPåSøkerinfo(søkerinfoMapper.mapTilSøkerinfo(personinfoDto, aktuelleBarn))
+    private fun relasjonerHarAdressesperreSøkerHarIkkeAdressesperre(
+            pdlSøker: PdlSøker,
+            aktuelleBarn: Map<String, PdlBarn>,
+            andreForeldre: Map<String, PdlAnnenForelder>
+    ): Boolean {
+
+        return when {
+            pdlSøker.harIkkeBeskyttetAdresse() && aktuelleBarn.harBeskyttetAdresse() -> true
+            pdlSøker.harIkkeBeskyttetAdresse() && andreForeldre.harBeskyttetAdresse() -> true
+            else -> false
+        }
+
     }
 
-    private fun harIkkeBeskyttetAdresse(adressebeskyttelse: List<Adressebeskyttelse>): Boolean {
-        return !kreverAdressebeskyttelse.contains(adressebeskyttelse.firstOrNull()?.gradering
-                                                  ?: AdressebeskyttelseGradering.UGRADERT)
+    private fun relasjonerHarStrengereBeskyttelseEnnSøker(
+            pdlSøker: PdlSøker,
+            aktuelleBarn: Map<String, PdlBarn>,
+            andreForeldre: Map<String, PdlAnnenForelder>
+    ): Boolean {
+        return when {
+            pdlSøker.harBeskyttetAdresseIkkeStreng() && aktuelleBarn.harStrengBeskyttetAdresse() -> true
+            pdlSøker.harBeskyttetAdresseIkkeStreng() && andreForeldre.harStrengBeskyttetAdresse() -> true
+            else -> false
+        }
+    }
+
+    override fun hentSøkerNavn(): String {
+        val søkersPersonIdent = EksternBrukerUtils.hentFnrFraToken()
+        val pdlSøker = pdlClient.hentSøker(søkersPersonIdent)
+        return pdlSøker.navn.first().visningsnavn()
     }
 
-    fun erILive(pdlBarn: PdlBarn) =
-            pdlBarn.dødsfall.firstOrNull()?.dødsdato == null
 
-    private fun settNavnFraPdlPåSøkerinfo(søkerinfo: Søkerinfo): Søkerinfo {
-        val pdlSøker = pdlClient.hentSøker(EksternBrukerUtils.hentFnrFraToken())
-        val søker = søkerinfo.søker
-        val mellomnavn = pdlSøker.navn.first().mellomnavn?.let { " $it " } ?: " "
-        val oppdaterSøker =
-                søker.copy(forkortetNavn = "${pdlSøker.navn.first().fornavn}$mellomnavn${pdlSøker.navn.first().etternavn}")
-        return søkerinfo.copy(søker = oppdaterSøker)
+    private fun hentAndreForeldre(
+            aktuelleBarn: Map<String, PdlBarn>,
+        søkersPersonIdent: String
+    ): Map<String, PdlAnnenForelder> {
+        return aktuelleBarn.map { it.value.familierelasjoner }
+            .flatten()
+            .filter { it.relatertPersonsIdent != søkersPersonIdent && it.relatertPersonsRolle != Familierelasjonsrolle.BARN }
+            .map { it.relatertPersonsIdent }
+            .distinct()
```

Comment:
Her burde vi filtrere bort søkeren, då vi ikke trenger å hente informasjon om den på nytt. 
Er nok noe vi burde gjøre i ef-sak også

Reply:
Gjøres ikke det med denne? it.relatertPersonsIdent != søkersPersonIdent 
 Skal se en gang til på det!

Reply:
Ah jo riktig! 👍